labview/labview_rabbitmq.c

```diff
@@ -157,99 +281,146 @@ int lv_amqp_exchange_declare(int64_t conn_intptr, uint16_t  channel, char *excha
 	tval.tv_usec = 0;
 
 	socket = amqp_tcp_socket_new(conn);
-	if (!socket) {
+	if (!socket)
+	{
 		return _CREATING_TCP_SOCKET;
-  	}
+	}
+
 	status = amqp_socket_open_noblock(socket, host, port, &tval);
-	if (status<0) {
-    return _OPENING_TCP_SOCKET;
-  	} 
+	if (status < 0)
+	{
+		return _OPENING_TCP_SOCKET;
+	}
+
 	/*Code explanation:
 	socket is set/stored in the connection state,
 	so we dont need to destroy this socket because
-	it will be destroyed along with connection state destroy function*/ 
-
-	char const *VHOST = "/"; 		// the virtual host to connect to on the broker. The default on most brokers is "/"
-	int const CHANNEL_MAX = 0; 		// the limit for number of channels for the connection. 0 means no limit.
-	int const FRAME_MAX = 131072; 	// the maximum size of an AMQP frame. 131072 is the default. 
-									// 4096 is the minimum size, 2^31-1 is the maximum, a good default is 131072 (128KB),
-	int const HEARTBEAT = 0; 		// the number of seconds between heartbeat frames to request of the broker. A value of 0 disables heartbeats.
-
-	return lv_report_amqp_error(amqp_login(conn, VHOST, CHANNEL_MAX, FRAME_MAX, HEARTBEAT, AMQP_SASL_METHOD_PLAIN, username, password), "Logging in", error_description);	
+	it will be destroyed along with connection state destroy function*/
+
+	char
+	const *VHOST = "/";	// the virtual host to connect to on the broker. The default on most brokers is "/"
+	int
+	const CHANNEL_MAX = 0;	// the limit for number of channels for the connection. 0 means no limit.
+	int
+	const FRAME_MAX = 131072;	// the maximum size of an AMQP frame. 131072 is the default. 
+	// 4096 is the minimum size, 2^31-1 is the maximum, a good default is 131072 (128KB),
+	int
+	const HEARTBEAT = 0;	// the number of seconds between heartbeat frames to request of the broker. A value of 0 disables heartbeats.
+
+	return lv_report_amqp_error(amqp_login(conn, VHOST, CHANNEL_MAX, FRAME_MAX, HEARTBEAT, AMQP_SASL_METHOD_PLAIN, username, password), "Logging in", error_description);
 }
 
 
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_basic_publish(int64_t conn_intptr, uint16_t  channel, char *exchange, char *routingkey, char *messagebody, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+int lv_amqp_basic_publish(int64_t conn_intptr, uint16_t channel, char *exchange, char *routingkey, char *cheaders_key, char *cheaders_value, char *messagebody, LStrHandle error_description)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
 	amqp_basic_properties_t props;
 	props._flags = AMQP_BASIC_CONTENT_TYPE_FLAG | AMQP_BASIC_DELIVERY_MODE_FLAG;
 	props.content_type = amqp_cstring_bytes("text/plain");
-	props.delivery_mode = 2; /* persistent delivery mode */
-	return amqp_basic_publish(conn, channel, amqp_cstring_bytes(exchange),amqp_cstring_bytes(routingkey), 0, 0, &props, amqp_cstring_bytes(messagebody));
+	props.delivery_mode = 2; /*persistent delivery mode */
+
+	int count, count2;
```

Comment:
if you wanna have separate strings, we should at least somewhere verify that count == count 2

Reply:
I think not, because it is nicely secured on the LabVIEW side, look how the caller creates those arrays:
![image](https://github.com/zaphiro-technologies/rabbitmq-c/assets/8245635/def67d4c-111f-4962-a5a1-0139c8ea0f2c)
there is no way that those arrays will have a different number of elements.

Reply:
but this may be a "creator" issue - as I know both sides of this interface. I'm not sure if I should look on it separately - because in that case the check that you pointed out - is required.